osf/models/node.py

```diff
@@ -2267,26 +2286,29 @@ def remove_node(self, auth, date=None):
                 '{0!r} does not have permission to modify this {1}, or a component in its hierarchy.'.format(auth.user, self.category or 'node')
             )
 
-        # After delete callback
-        remove_addons(auth, hierarchy)
+        with transaction.atomic():
+            # After delete callback
+            remove_addons(auth, hierarchy)
 
-        Comment = apps.get_model('osf.Comment')
-        Comment.objects.filter(node__id__in=hierarchy).update(root_target=None)
+            Comment = apps.get_model('osf.Comment')
+            Comment.objects.filter(node__id__in=hierarchy).update(root_target=None)
 
-        log_date = date or timezone.now()
-        for node in hierarchy:
-            # Add log to parents
-            node.is_deleted = True
-            node.deleted_date = date
-            node.add_remove_node_log(auth=auth, date=log_date)
-            project_signals.node_deleted.send(node)
+            log_date = date or timezone.now()
+            for node in hierarchy:
+                node.is_deleted = True
+                node.deleted_date = date
+                # Deletes OSFStorage root folder; Contents deleted separately in celery task.
+                node.root_folder.delete(recursive=False)
```

Comment:
Two changes to `Node.remove_node()`. 1) Removal wrapped in a transaction and 2) root_folder is trashed here, instead of waiting for the celery task to delete this.